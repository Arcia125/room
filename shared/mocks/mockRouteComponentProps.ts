/**
 * @description Mock props for Children of Route components.
 */
import { RouteComponentProps, match } from 'react-router';
import * as history from 'history';

const mockLocation: history.Location = {
  hash: '',
  pathname: '/',
  search: '/',
  state: {},
};

const mockHistory: history.History = {
  push: jest.fn(),
  length: 1,
  block: jest.fn(),
  action: 'PUSH',
  createHref: jest.fn(),
  go: jest.fn(),
  goBack: jest.fn(),
  goForward: jest.fn(),
  listen: jest.fn(),
  location: mockLocation,
  replace: jest.fn(),
};

const mockMatch: match = {
  isExact: true,
  params: {},
  path: '/',
  url: '/',
};

/**
 * @description mocked version of props passed from react router Routes to their children
 * @example
 * render(<MyComponent {...mockRouteComponentProps} />)
 */
export const mockRouteComponentProps: RouteComponentProps = {
  history: mockHistory,
  location: mockLocation,
  match: mockMatch,
};
