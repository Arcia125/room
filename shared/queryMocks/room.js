import * as generate from '../mockData/generate';
import { GET_ROOM } from '../../src/graphql/getRoom';
import { NEW_ROOM_MESSAGE } from '../../src/graphql/newRoomMessage';
import { users } from '../mockData/users';
import { NEW_ROOM_USER } from '../../src/graphql/newRoomUser';
import { JOIN_ROOM } from '../../src/graphql/joinRoom';
import { ADD_ROOM } from '../../src/graphql/addRoom';

const mockRoom = generate.room({
  id: 'test-id',
  name: 'test-room',
  messages: generate.messages(2),
  users,
});

const { id: roomId } = mockRoom;

const getRoomMock = {
  request: {
    query: GET_ROOM,
    variables: { id: roomId },
  },
  result: {
    data: {
      room: mockRoom,
    },
  },
};

const newRoomMessageMock = {
  request: {
    query: NEW_ROOM_MESSAGE,
    variables: { roomId },
  },
  result: {
    data: {
      newRoomMessage: generate.message(),
    },
  },
};

const newRoomUserMock = {
  request: {
    query: NEW_ROOM_USER,
    variables: { roomId },
  },
  result: {
    data: {
      newRoomUser: generate.user(),
    },
  },
};

const joinRoomMock = {
  request: {
    query: JOIN_ROOM,
    variables: { roomId },
  },
  result: {
    data: {
      joinRoom: {
        success: true,
        __typename: 'SuccessResult',
      },
    },
  },
};

const addRoomMock = {
  request: {
    query: ADD_ROOM,
  },
  result: {
    data: {
      addRoom: generate.room(),
    },
  },
};

export {
  mockRoom,
  getRoomMock,
  newRoomMessageMock,
  newRoomUserMock,
  joinRoomMock,
  addRoomMock,
};
