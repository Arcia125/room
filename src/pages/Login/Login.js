import React from 'react';
import PropTypes from 'prop-types';

import { LoginForm } from '../../components/LoginForm';
import { useLoginForm } from '../../hooks/useLoginForm';
import { useAuth } from '../../hooks/useAuth';

const Login = props => {
  const auth = useAuth();

  const {
    username,
    password,
    onChangeUsername,
    onChangePassword
  } = useLoginForm();

  const onSubmit = e => {
    e.preventDefault();
    auth.loginUser({ username, password });
  };

  return (
    <div>
      <LoginForm
        onSubmit={onSubmit}
        username={username}
        onChangeUsername={onChangeUsername}
        password={password}
        onChangePassword={onChangePassword}
      />
    </div>
  );
};

Login.propTypes = {};

export default Login;
