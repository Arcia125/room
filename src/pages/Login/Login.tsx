import React, { FunctionComponent } from 'react';
import { RouteComponentProps } from 'react-router';

import { LoginForm } from '../../components/LoginForm';
import { useLoginForm } from '../../hooks/useLoginForm';
import { useAuth } from '../../hooks/useAuth';
import styled from 'styled-components';
import Navbar from '../../components/Navbar';

const StyledLogin = styled.div`
  background-color: ${p => p.theme.colors.lighterPurple};
  min-height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  .loginPage {
    display: grid;
    grid-gap: 32px;
    width: 348px;
    max-width: 90vw;
    &__invite {
      font-family: 'Poppins', sans-serif;
      font-size: 2.58rem;
      line-height: 1.5;
      text-align: center;
    }
  }
`;

const Login: FunctionComponent<RouteComponentProps> = ({ history }) => {
  const auth = useAuth();

  const {
    username,
    password,
    onChangeUsername,
    onChangePassword,
  } = useLoginForm();

  if (auth.currentUser) {
    // if there is a current user, redirect to dashboard
    history.push('/dashboard');
  }

  const onSubmit: React.FormEventHandler<HTMLFormElement> = event => {
    event.preventDefault();
    auth.loginUser({ username, password });
  };

  return (
    <StyledLogin>
      <Navbar />
      <div className="loginPage">
        <p className="loginPage__invite">Come back in!</p>
        <LoginForm
          onSubmit={onSubmit}
          username={username}
          onChangeUsername={onChangeUsername}
          password={password}
          onChangePassword={onChangePassword}
        />
      </div>
    </StyledLogin>
  );
};

Login.propTypes = {};

export default Login;
