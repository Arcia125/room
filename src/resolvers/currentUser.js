import { getUser, setUser, getToken } from '../utils/user';
// let user = {
//   id: null,
//   username: null,
//   __typename: 'User'
// };

// let user = getUser();

const currentUser = {
  defaults: {
    currentUser: getUser()
  },
  resolvers: {
    Mutation: {
      saveUser: (_, { username, id, email, token }, { cache }) => {
        const user = { id, username, email, __typename: 'User' };
        console.log('saveUser resolver', user, token);
        // console.log('saving user ', user);
        // cache.writeQuery({
        //   query: GET_CURRENT_USER,
        //   data: { currentUser: user },
        // });

        setUser({ user, cache, token });

        return { user, token, __typename: 'AuthPayload' };
        // cache.writeData({
        //   currentUser: {
        //     id,
        //     username,
        //     __typename: 'User',
        //   },
        // });
      }
    },
    Query: {
      currentUser: (root, args, { cache }, info) => {
        // debugger;
        console.log('getting current user');
        // const queryResult = cache.readQuery({
        //   query: GET_CURRENT_USER,
        // });
        // console.log('readQuery', queryResult);
        const user = getUser();
        return user ? user : null;
        // return {
        //   user: getUser(),
        //   token: getToken(),
        // };
      }
    }
  }
};

export { currentUser };
