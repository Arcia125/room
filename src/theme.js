const theme = {
  spacing: {
    0: '0px',
    1: '8px',
    2: '16px',
    3: '32px'
  }
};

export { theme };
