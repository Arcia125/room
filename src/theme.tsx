import React, { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  spacing: {
    0: '0px',
    1: '8px',
    2: '16px',
    3: '32px',
  },
  colors: {
    gray: '#B3B2C0',
    grayDark: '#1f1f1f',
    lightBlue: '#AEAFE8',
    blue: '#AC2FC9',
    lightPurple: '#EFEEF8',
    purple: '#9056D4',
    pink: '#DD6589',
    bleh: '#8A4966',
  },
};

const Provider: React.FunctionComponent<{ children: ReactElement }> = ({
  children,
  ...restProps
}) => (
  <ThemeProvider theme={theme} {...restProps}>
    {children}
  </ThemeProvider>
);

export { theme, Provider };
