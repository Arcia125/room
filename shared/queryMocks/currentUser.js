import { users } from '../mockData/users';
import { GET_CURRENT_USER } from '../../src/graphql/getCurrentUser';
import { currentUser } from '../../src/resolvers/currentUser';


const mockUser = users[0];

const currentUserMock = {
  request: {
    query: GET_CURRENT_USER,
  },
  result: {
    data: {
      currentUser: mockUser
    }
  }
};

const currentUserMockResolvers = {
  Mutation: currentUser.resolvers.Mutation,
  Query: {
    currentUser: () => mockUser
  }
};

const nullCurrentUserMock = {
  request: {
    query: GET_CURRENT_USER,
  },
  result: {
    data: {
      currentUser: null
    }
  }
};

const nullCurrentUserMockResolvers = {
  Mutation: currentUser.resolvers.Mutation,
  Query: {
    currentUser: () => null
  }
};



export { currentUserMock, currentUserMockResolvers, nullCurrentUserMock, nullCurrentUserMockResolvers };