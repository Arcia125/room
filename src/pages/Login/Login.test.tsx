import React from 'react';
import { MockedProvider } from '@apollo/react-testing';
import { render, cleanup, wait } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router';

import Login from '.';
import { Provider } from '../../theme';
import { mockRouteComponentProps } from '../../../shared/mocks/mockRouteComponentProps';
import { nullCurrentUserMockResolvers } from '../../../shared/queryMocks/currentUser';

afterEach(cleanup);

describe('Login', () => {
  it('renders without crashing', async () => {
    const { container, asFragment } = render(
      <MemoryRouter>
        <MockedProvider addTypename resolvers={nullCurrentUserMockResolvers}>
          <Provider>
            <Login {...mockRouteComponentProps} />
          </Provider>
        </MockedProvider>
      </MemoryRouter>
    );

    await wait(() => expect(container).toBeInTheDocument());

    expect(asFragment()).toMatchSnapshot();
  });
});
