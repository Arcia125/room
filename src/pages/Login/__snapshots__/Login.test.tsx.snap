// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-bxivhb ebgYGz"
  >
    <nav
      class="sc-htpNat gHSSDV"
    >
      <a
        href="/"
      >
        <img
          alt="Room Logo"
          src="logoText.svg"
        />
      </a>
    </nav>
    <div
      class="loginPage"
    >
      <p
        class="loginPage__invite"
      >
        Come back in!
      </p>
      <form
        class="loginForm sc-bwzfXH iLgZzU"
      >
        <div
          class="loginForm__inputs"
        >
          <div
            class="loginForm__input"
          >
            <label
              for="username"
            >
              Username
            </label>
            <input
              id="username"
              type="text"
              value=""
            />
          </div>
          <div
            class="loginForm__input"
          >
            <label
              for="password"
            >
              Password
            </label>
            <input
              id="password"
              type="password"
              value=""
            />
            <a
              class="loginForm__helperText"
              href="/forgot-password"
            >
              Forgot your password?
            </a>
          </div>
        </div>
        <div
          class="loginForm__submit"
        >
          <button
            class="sc-bdVaJa iGgGiy"
            color="primary"
            type="submit"
          >
            login
          </button>
          <p
            class="loginForm__helperText"
          >
            <span>
              Need an account?
            </span>
            <a
              href="/"
            >
              Register
            </a>
          </p>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
